docs/sql-ref-datetime-pattern.md

```diff
@@ -189,6 +209,18 @@ Spark uses the below letters in date and timestamp parsing and formatting:
   <td> literal </td>
   <td> ' </td>
 </tr>
+<tr>
+  <td> <b>[</b> </td>
+  <td> optional section start </td>
```

Comment:
shall we further expalin it in the detail section below?

Reply:
explanations added